src/components/panningViews/panToDismissView.js

```diff
@@ -0,0 +1,301 @@
+    if (this.swipe[0] || this.swipe[1]) {
+      const {isRight, isDown} = this.getDismissAnimationDirection();
+      this.animateDismiss(isRight, isDown);
+    } else {
+      const endValue = [Math.round(this.left), Math.round(this.top)];
+      if (
+        (directions.includes(PanningProvider.Directions.LEFT) && endValue[0] <= -this.thresholdX) ||
+        (directions.includes(PanningProvider.Directions.RIGHT) && endValue[0] >= this.thresholdX) ||
+        (directions.includes(PanningProvider.Directions.UP) && endValue[1] <= -this.thresholdY) ||
+        (directions.includes(PanningProvider.Directions.DOWN) && endValue[1] >= this.thresholdY)
+      ) {
+        const {isRight, isDown} = this.getDismissAnimationDirection();
+        this.animateDismiss(isRight, isDown);
+      } else {
+        this.animateToInitialPosition();
+      }
+    }
+  };
+
+  animateToInitialPosition = () => {
+    const {deltaX, deltaY} = this.state;
+    const {animationSpeed, animationBounciness} = this.props;
+    const toX = -this.left;
+    const toY = -this.top;
+    const animations = [];
+    if (!_.isUndefined(toX)) {
+      animations.push(
+        Animated.spring(deltaX, {
+          toValue: Math.round(toX),
+          speed: animationSpeed,
+          bounciness: animationBounciness,
+        }),
+      );
+    }
+
+    if (!_.isUndefined(toY)) {
+      animations.push(
+        Animated.spring(deltaY, {
+          toValue: Math.round(toY),
+          speed: animationSpeed,
+          bounciness: animationBounciness,
+        }),
+      );
+    }
+
+    this.setState({shouldAnimate: true}, () => {
+      Animated.parallel(animations).start(this.onInitAnimationFinished);
+    });
+  };
+
+  onInitAnimationFinished = () => {
+    this.setState({shouldAnimate: false});
+    this.initPositions();
+  };
+
+  getDismissAnimationDirection = () => {
+    const {swipeDirections, dragDirections} = this.props.context;
+    const hasHorizontalSwipe = !_.isUndefined(swipeDirections[0]);
+    const hasVerticalSwipe = !_.isUndefined(swipeDirections[1]);
+    let isRight;
+    let isDown;
+    
+    if (hasHorizontalSwipe || hasVerticalSwipe) {
+      if (hasHorizontalSwipe) {
+        isRight = swipeDirections[0] === PanningProvider.Directions.RIGHT;
+      }
+
+      if (hasVerticalSwipe) {
+        isDown = swipeDirections[1] === PanningProvider.Directions.DOWN;
+      }
+    } else {
+      // got here from a drag beyond threshold
+      if (!_.isUndefined(dragDirections[0])) {
+        isRight = dragDirections[0] === PanningProvider.Directions.RIGHT;
+      }
+
+      if (!_.isUndefined(dragDirections[1])) {
+        isDown = dragDirections[1] === PanningProvider.Directions.DOWN;
+      }
+    }
+
+    return {isRight, isDown};
+  }
+
+  animateDismiss = (isRight, isDown) => {
+    const {deltaX, deltaY} = this.state;
+    const animations = [];
+    let toX;
+    let toY;
+
+    if (!_.isUndefined(isRight) && !_.isUndefined(deltaX)) {
```

Comment:
unless you really want to check if a value is `undefined` and just check that it's "falsy", writing `!isRight` for instance is good enough (:

Reply:
I know, but I had a few bugs with that so I started using _.isUndefined more often (unless it's a real boolean).
Is that OK?
Specifically here, I'm unsure how the animation value (0) will behave and the isRight can be true\false\undefined.

Reply:
it's possible `deltaX` is always truthy (:

Reply:
Done